lib/vrl/compiler/src/expression/function_call.rs

```diff
@@ -382,7 +391,16 @@ impl Expression for FunctionCall {
             None => return Err(format!("Function {} not found.", self.function_id)),
         };
 
-        let compile_ctx = FunctionCompileContext { span: self.span };
+        // We take the external context, and pass it to the function compile context, this allows
+        // functions mutable access to external state, but keeps the internal compiler state behind
+        // an immutable reference, to ensure compiler state correctness.
+        //
+        // FIXME: need mutable access to compiler state here, but the VM implementation doesn't
+        // allow that yet...
+        // let external_context = state.swap_external_contexts(AnyMap::new());
```

Comment:
@StephenWakely could you help me figure out a way how to make this work for the new VM implementation?